cataclysmbnteam/Cataclysm-BN
# Add mountable autodoc and autodoc couch (with EgorBelphegor additions)

#### Summary
SUMMARY: Content "Add mountable autodoc and autodoc couch"

#### Purpose of change
There is already an autoclave to sterilize CBMs, but you still had to transport your autodoc and your autodoc couch as items, and installing/removing them each time you need an autodoc.

#### Describe the solution
Add the mountable autodoc/autodoc couch items to the game.
They can do the same things as their furniture counterpart (cf tests).
They don't use power, same as their furniture counterpart.

Edit: Since @EgorBelphegor proposed an other [PR](https://github.com/cataclysmbnteam/Cataclysm-BN/pull/2140) for this feature, I merged a part of it in this one.

The most notable changes are:
- you can now operate while the vehicle is moving (even in an helicopter, and going over bridges is supported too)
- the code is a bit cleaner, I especially liked the usage of a function to check if a flag is on a furniture or a vehicle part, this could be useful in the future. I hesitated to add it when I was developping the PR, but seeing the result it is indeed much better.
- hostile nursebots are able to use autodoc vehicles too. I didn't bother to add it because the chance of this happening seemed very slim (they can only remove ~2 bionics, and they can't reload their anesthetic after that), but it's better if they can do it noneless.

So even if @EgorBelphegor probably didn't see this PR, it was still very useful, thanks.


#### Testing
- install CBM
- uninstall CBM
- splint broken limb (can splint arms and legs of people with normal sized body, same as the furniture version)
- treat wounds (can stop bleeding and other almost hidden features, same as the furniture version)
- check radiation level
- do everything above on a follower
- convert a broken cyborg (failure and success)
- repeat every actions above, but on a normal autodoc
- craft a mountable autodoc/autodoc couch

Edit: I retested everything above since the integration of @EgorBelphegor's PR
